sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -49,50 +48,58 @@
 class SetStatementAnalyzer {
 
     private static final Logger LOGGER = LogManager.getLogger(SetStatementAnalyzer.class);
+    private final Functions functions;
 
-    public static AnalyzedStatement analyze(SetStatement node) {
+    public SetStatementAnalyzer(Functions functions) {
+        this.functions = functions;
+    }
+
+    public AnalyzedStatement analyze(SetStatement<Expression> node,
+                                     ParamTypeHints typeHints,
+                                     CoordinatorTxnCtx txnCtx) {
 
         boolean isPersistent = node.settingType().equals(SetStatement.SettingType.PERSISTENT);
-        Map<String, List<Expression>> settings = new HashMap<>();
-        Assignment<Expression> assignment;
+        Map<String, List<Symbol>> settings = new HashMap<>();
+
+        var exprAnalyzer = new ExpressionAnalyzer(
+            functions,
+            txnCtx,
+            typeHints,
+            FieldProvider.FIELDS_AS_LITERAL,
+            null
+        );
+
+        SetStatement<Symbol> statement = node.map(x -> exprAnalyzer.convert(x, new ExpressionAnalysisContext()));
 
         switch (node.scope()) {
             case LICENSE:
-                if (node.assignments().size() != SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM) {
+                if (node.assignments().size() != AnalyzedSetLicenseStatement.LICENSE_TOKEN_NUM) {
                     throw new IllegalArgumentException("Invalid number of arguments for SET LICENSE. " +
                                                        "Please provide only the license key");
                 }
-                assignment = node.assignments().get(0);
-                String licenseKey = ExpressionToStringVisitor.convert(assignment.expression(), Row.EMPTY);
-
-                return new SetLicenseAnalyzedStatement(licenseKey);
+                return new AnalyzedSetLicenseStatement(statement.assignments().get(0).expression());
             case GLOBAL:
-                for (Assignment<Expression> anAssignment : node.assignments()) {
-                    for (String setting : ExpressionToSettingNameListVisitor.convert(anAssignment)) {
-                        CrateSettings.checkIfRuntimeSetting(setting);
-                    }
-                    String settingName = ExpressionToStringVisitor.convert(anAssignment.columnName(), Row.EMPTY);
-                    settings.put(settingName, ImmutableList.of(anAssignment.expression()));
+                for (Assignment<Symbol> a : statement.assignments()) {
+                    String settingName = a.columnName().representation();
```

Comment:
Using the representation() isn't valid here. Every symbol should be evaluated after binding.